src/generic/generic-component-form-select/generic-component-form-select.scss

```diff
@@ -3,67 +3,92 @@
  * @define select
  */
 
-@import 'mixins';
+// Import base
+@import '@ecl/generic-base/generic-base';
 
-.ecl-select {
-  @include ecl-select-list-arrow($ecl-color-shade);
+// Variables
+$ecl-select-icon-size: map-get($ecl-font-size, 's');
+$ecl-select-background-width: $ecl-select-icon-size + 2 *
+  map-get($ecl-spacing, 'xxs');
 
-  appearance: none;
-  background-color: #fff;
-  border: 1px solid $ecl-color-shade;
-  border-radius: 0;
-  color: $ecl-color-shade;
-  display: block;
-  font-family: $ecl-font-family-sans-serif;
-  font-size: map-get($ecl-font-size, 's');
-  margin: 0;
-  padding: map-get($ecl-spacing, 'xxs')
-    ($ecl-select-background-width + map-get($ecl-spacing, 'xxxs'))
-    map-get($ecl-spacing, 'xxs') map-get($ecl-spacing, 'xxs');
-  width: 100%;
+@mixin ecl-select() {
+  .ecl-select {
+    @include ecl-select-list-arrow($ecl-color-shade);
```

Comment:
`ecl-select-list-arrow` is used before it's defined, this is not a good practice (readability)